src/app/utils/logic-adaptor.ts

```diff
@@ -1,32 +1,126 @@
-import { ok, Result } from 'neverthrow'
+import { cloneDeep, omit } from 'lodash'
+import { err, fromThrowable, ok, Result } from 'neverthrow'
 
+import { hasProp } from '../../shared/util/has-prop'
 import {
   FieldIdSet,
   getLogicUnitPreventingSubmit as logicGetLogicUnitPreventingSubmit,
   getVisibleFieldIds as logicGetVisibleFieldIds,
```

Comment:
hmm this is a very non-intuitive way of namespacing the logic module. appending the word `logic` in front doesn't make it clear that this comes from the logic module. can we do an `import *` instead?